src/Adapter/OrderReturn/Repository/OrderReturnRepository.php

```diff
@@ -86,4 +110,119 @@ public function update(OrderReturn $orderReturn): void
             OrderReturnException::class
         );
     }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     *
+     * @throws CoreException
+     * @throws DeleteOrderReturnProductException
+     * @throws OrderReturnException
+     */
+    public function deleteOrderReturnProduct(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId
+    ): void {
+        $orderReturn = $this->get($orderReturnId);
+
+        if ((int) ($orderReturn->countProduct()) <= 1) {
+            throw new DeleteOrderReturnProductException(
+                'Can\'t delete last product from merchandise return',
+                DeleteOrderReturnProductException::LAST_ORDER_RETURN_PRODUCT
+            );
+        }
+        $orderReturnDetail = $this->getOrderReturnDetailByOrderDetailId($orderReturnDetailId->getValue());
+
+        $this->deleteOrderReturnDetail(
+            $orderReturnDetail->getOrderReturnId(),
+            $orderReturnDetailId,
+            $orderReturnDetail->getCustomizationId()
+        );
+    }
+
+    /**
+     * @param OrderReturnId $orderReturnId
+     * @param OrderReturnDetailId $orderReturnDetailId
+     * @param CustomizationId|null $customizationId
+     *
+     * @throws DeleteOrderReturnProductException
+     */
+    public function deleteOrderReturnDetail(
+        OrderReturnId $orderReturnId,
+        OrderReturnDetailId $orderReturnDetailId,
+        ?CustomizationId $customizationId = null
+    ): void {
+        try {
+            if (!OrderReturn::deleteOrderReturnDetail(
+                $orderReturnId->getValue(),
+                $orderReturnDetailId->getValue(),
+                $customizationId ? $customizationId->getValue() : 0
+            )) {
+                throw new DeleteOrderReturnProductException(
+                    'Failed to delete merchandise return detail',
+                    DeleteOrderReturnProductException::UNEXPECTED_ERROR
+                );
+            }
+        } catch (PrestaShopException $e) {
```

Comment:
Maybe you can use another Exception than the legacy one. So we don't add more legacy usages.

Reply:
Well this catch here is in case there is a legacy exception to throw a core exception in this case DeleteOrderReturnProductException

Reply:
ok